storage/tianmu/core/tianmu_attr_exqp.cpp

```diff
@@ -922,20 +932,20 @@ void TianmuAttr::EvaluatePack_BetweenInt(MIUpdatingIterator &mit, int dim, Descr
       if (d.op == common::Operator::O_BETWEEN && !mit.NullsPossibleInPack(dim) && dpn.numOfNulls == 0) {
         // easy and fast case - no "if"s
         do {
-          auto v = p->GetValInt(mit.GetCurInpack(dim));
-          if (pv1 > v || v > pv2)
+          uint64_t v = p->GetValInt(mit.GetCurInpack(dim));
+          if (empty_flag || v < upv1 || v > upv2)
```

Comment:
If you use `empty_flag`,  i would recommend you to using `(pv2 < local_min)` instead, that keeps same `if` statement.